markdowns/20_Using_Haystack_with_REST_API.md

```diff
@@ -0,0 +1,312 @@
+---
+layout: tutorial
+colab: False
+toc: True
+title: "Using Haystack with REST API"
+last_updated: 2023-01-04
+level: "advanced"
+weight: 115
+description: Create a production-ready pipeline and interact with Haystack REST API.
+category: "QA"
+aliases: ['/tutorials/using-haystack-with-rest-api']
+download: "/downloads/20_Using_Haystack_with_REST_API.ipynb"
+---
+    
+
+
+- **Level**: Advanced
+- **Time to complete**: 30 minutes
+- **Prerequisites**: Basic understanding of Docker, basic knowledge of Haystack pipelines 
+- **Nodes Used**: `ElasticsearchDocumentStore`, `EmbeddingRetriever`
+- **Goal**: After completing this tutorial, you will have learned how to create a pipeline YAML file, index files, and how to query your pipeline using REST API.
+
+## Overview
+
+Learn how you can interact with Haystack through REST API. This tutorial introduces you to all the concepts needed to build an end-to-end document search application.  
+
+With Haystack, you can apply the latest NLP technology to your own data and create production-ready applications. Building an end-to-end NLP application requires the combination of multiple concepts:
+* **DocumentStore** is the component in Haystack responsible for loading and storing text data in the form of [Documents](https://docs.haystack.deepset.ai/docs/documents_answers_labels#document). In this tutorial, the DocumentStore uses Elasticsearch behind the scene.
+* **Haystack pipelines** convert files into Documents, index them to the DocumentStore, and run NLP tasks such as question answering and document search.
+* **REST API**, as a concept, makes it possible for applications to interact with each other by handling their queries and returning responses. There is `rest_api` application within Haystack that exposes Haystack's functionalities through a RESTful API.
+* **Docker** simplifies the environment setup needed to run Elasticsearch and Haystack API.
+
+
+
+## Preparing the Environment
+
+1. Install [Docker Compose](https://docs.docker.com/compose/) and launch Docker.
+If you installed Docker Desktop, just start the application. Run `docker info` to see if Docker is up and running:
+
+   ```bash
+   docker info
+   ```
+
+2. Download the *docker-compose.yml* file. Haystack provides a *docker-compose.yml* file that defines services for Haystack API and Elasticsearch. 
+    1. Create a new folder called *doc-search*.
```

Comment:
Cosmetic, not too important so take or leave this comment: 
As a developer my first question might be 'where'? So maybe somewhere above I would mention that these are the steps for a brand new project we have set up.